crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -474,7 +474,10 @@ where
         //    and deemed `VALID`. In the case of such an event, client software MUST return
         //    `{payloadStatus: {status: VALID, latestValidHash: forkchoiceState.headBlockHash,
         //    validationError: null}, payloadId: null}`
-        attrs.take();
+        if self.blockchain.canonical_tip() != header.num_hash() {
```

Comment:
I feel something is missing here. Dont take Attributs only if `header` is equal to `canonical_tip`, but what if the header is past or future block? And should this be `header.parent` is `canonical_tip`?

Reply:
> what if the header is past or future block

then the payload building process should be skipped 

> And should this be header.parent is canonical_tip

no

Reply:
talked in chat, lgtm